test/Microsoft.Azure.WebJobs.Extensions.Kafka.EndToEndTests/KafkaEndToEndTestFixture.cs

```diff
@@ -0,0 +1,103 @@
+
+        internal TopicSpecification StringTopicWithOnePartition { get; } = new TopicSpecification() { Name = Constants.StringTopicWithOnePartitionName, NumPartitions = 1, ReplicationFactor = 1 };
+
+        internal TopicSpecification StringTopicWithTenPartitions { get; } = new TopicSpecification() { Name = Constants.StringTopicWithTenPartitionsName, NumPartitions = 10, ReplicationFactor = 1 };
+
+        internal TopicSpecification StringTopicWithLongKeyAndTenPartitions { get; } = new TopicSpecification() { Name = Constants.StringTopicWithLongKeyAndTenPartitionsName, NumPartitions = 10, ReplicationFactor = 1 };
+
+        internal TopicSpecification MyAvroRecordTopic { get; } = new TopicSpecification() { Name = Constants.MyAvroRecordTopicName, NumPartitions = 10, ReplicationFactor = 1 };
+
+        internal TopicSpecification MyProtobufTopic { get; } = new TopicSpecification() { Name = Constants.MyProtobufTopicName, NumPartitions = 10, ReplicationFactor = 1 };
+
+        public KafkaEndToEndTestFixture()
+        {
+            var config = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.tests.json", optional: true)
+                .AddJsonFile("appsettings.tests.local.json", optional: true)
+                .Build();
+
+            var brokerFromConfig = config["Values:LocalBroker"];
+            if (!string.IsNullOrWhiteSpace(brokerFromConfig))
+            {
+                this.Broker = brokerFromConfig;
+            }
+        }
+
+
+        /// <summary>
+        /// Helper to get all topics used in the end to end tests
+        /// </summary>
+        /// <returns>The all topics.</returns>
+        public IEnumerable<TopicSpecification> GetAllTopics()
+        {
+            foreach (var prop in this.GetType().GetProperties().Where(x => x.PropertyType == typeof(TopicSpecification)))
+            {
+                yield return (TopicSpecification)prop.GetValue(this);
+            }
+        }
+
+        public Task DisposeAsync() => Task.CompletedTask;
+
+        /// <summary>
+        /// Creates the required topics for end to end tests
+        /// </summary>
+        /// <returns>The async.</returns>
+        public async Task InitializeAsync()
+        {
+            var adminClientBuilder = new AdminClientBuilder(new AdminClientConfig()
+            {
+                BootstrapServers = this.Broker,
+
+            });
+
+            var adminClient = adminClientBuilder.Build();
+            try
+            {
+                var createTopicOptions = new CreateTopicsOptions()
+                {
+                    OperationTimeout = TimeSpan.FromMinutes(2),
+                    RequestTimeout = TimeSpan.FromMinutes(2),
+                };
+
+                await adminClient.CreateTopicsAsync(GetAllTopics(), createTopicOptions);
+            }
+            catch (KafkaException ex)
+            {
+                if (ex is CreateTopicsException createTopicsException)
```

Comment:
Could you put this in a `catch (CreateTopicsException)` instead of switching in here?